lua/entities/gmod_wire_expression2/core/extpp.lua

```diff
@@ -228,279 +149,155 @@ function E2Lib.ExtPP.Pass2(contents)
 	-- This flag helps determine whether the preprocessor changed, so we can tell the environment about it.
 	local changed = false
 	for a_begin, attributes, h_begin, ret, thistype, colon, name, args, whitespace, equals, h_end in contents:gmatch("()(%[?[%w,_ =\"]*%]?)[\r\n\t ]*()e2function%s+(" .. p_typename .. ")%s+([a-z0-9]-)%s*(:?)%s*(" .. p_func_operator .. ")%(([^)]*)%)(%s*)(=?)()") do
-		-- Convert attributes to a lookup table passed to registerFunction
-		attributes = attributes ~= "" and attributes or nil
-		-- attributes = attributes ~= ""
-		local attributes_str
-
-		if attributes and attributes:sub(1, 1) == "[" and attributes:sub(-1, -1) == "]" then
-			attributes_str = attributes
-			attributes = attributes:sub(2, -2) -- Remove surrounding brackets
-			-- [deprecated, nodiscard]
-			-- e2function void test()
-
-			attributes = attributes:Split(",")
-
-			local lookup = {}
-			for _, tag in ipairs(attributes) do
-				local k = tag:lower():Trim()
-
-				if k:find("=", 1, true) then
-					-- [xyz = 567, event = "Tick"]
-					-- e2function number foo()
-					local key, value = unpack( k:Split("="), 1, 2 )
-					key, value = key:lower():Trim(), tonumber(value:match("%d+")) or value:Trim()
-
-					lookup[key] = value
-				else
-					if not valid_attributes[k] then
-						ErrorNoHalt("Invalid attribute fed to ExtPP: " .. k)
-					end
-					lookup[ tag:lower():Trim() ] = "true"
-				end
-			end
-
-			local buf = "{"
-			for attr, val in pairs(lookup) do
-				buf = buf .. "['" .. attr .. "'] = " .. val .. ","
-			end
-
-			attributes = buf .. "}"
+		local line = select(2, contents:sub(1, h_begin):gsub("\n", ""))
+		local trace = "(at line " .. line .. ")"
+		local trace_ext = trace .. (E2Lib.currentextension and (" @" .. filename) or "")
+
+		if contents:sub(h_begin - 1, h_begin - 1):match("%w") then
+			ErrorNoHalt("Warning: Malformed e2function must not have characters before 'e2function' " .. trace_ext .. "\n")
+		elseif RemovedOperators[name] then -- Old operator that no longer is needed.
+			ErrorNoHalt("Warning: Operator " .. name .. " is now redundant. Ignoring registration. " .. trace_ext .. "\n")
```

Comment:
Where do these errors end up? server console?

Reply:
ErrorNoHalt is essentially just Msg, so yeah. Intended to bug server owners who then bug developers to remove that redundant code

Reply:
Will they bug wiremod developers or will they bug the extension developers?

Reply:
addressed in new post => resolving